### DISCUSSION TITLE
rd_kafka_metadata defines both struct and function

### DISCUSSION THREAD

**Discussion by baumgold** [edenhill/librdkafka#3829] [Q&A]
The `rd_kafka_metadata` name in [rdkafka.h](https://github.com/edenhill/librdkafka/blob/v1.8.2/src/rdkafka.h#L4574-L4614) seems to define both a struct and a function, which causes problems ([here](https://github.com/JuliaInterop/Clang.jl/blob/v0.15.8/src/generator/audit.jl#L76)) when trying to use Clang.jl to wrap the c-library into a Julia library.  I believe we should modify the struct definition such that `rd_kafka_metadata` is changed to `rd_kafka_metadata_s` to be consistent with typedef in other parts of the codebase..  Does this make sense?  If so I can create an issue/PR to make the change.  Thanks!

```c
typedef struct rd_kafka_metadata_s {
        int         broker_cnt;     /**< Number of brokers in \p brokers */
        struct rd_kafka_metadata_broker *brokers;  /**< Brokers */

        int         topic_cnt;      /**< Number of topics in \p topics */
        struct rd_kafka_metadata_topic *topics;    /**< Topics */

        int32_t     orig_broker_id;   /**< Broker originating this metadata */
        char       *orig_broker_name; /**< Name of originating broker */
} rd_kafka_metadata_t;
```

**Comment by edenhill** [comment]
Yeah that was a mistake not to name it rd_kafka_metadata_s, but we can't really change it now since it would break backwards compatibility with existing applications that are using the struct rather than the typedef.

But ultimately I think this is a problem with Clang.jl.
